src/main/java/com/questhelper/quests/thedepthsofdespair/TheDepthsOfDespair.java

```diff
@@ -0,0 +1,280 @@
+import com.questhelper.steps.ObjectStep;
+import com.questhelper.steps.QuestStep;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import net.runelite.api.Favour;
+import net.runelite.api.ItemID;
+import net.runelite.api.NpcID;
+import net.runelite.api.ObjectID;
+import net.runelite.api.QuestState;
+import net.runelite.api.Skill;
+import net.runelite.api.coords.WorldPoint;
+
+@QuestDescriptor(
+	quest = QuestHelperQuest.THE_DEPTHS_OF_DESPAIR
+)
+public class TheDepthsOfDespair extends BasicQuestHelper
+{
+	// Recommended
+	ItemRequirement weapon, dramenStaff, foodIfLowLevel, superEnergyOrStamina, xericsTalisman, skillsNecklace,
+		varlamoreEnvoy, royalAccordOfTwill;
+
+	ConditionForStep inVineryHouse, inArceuusLibrary, hasTheVarlamoreEnvoy, inCaves, inFirstAreaOfCaves,
+		inSecondAreaOfCaves, inThirdAreaOfCaves, downstairsInCaves;
+
+	QuestStep talkToLordKandur, talkToChefOlivia, talkToGalana, findTheVarlamoreEnvoy, readTheVarlamoreEnvoy,
+		enterCrabclawCaves, goThroughCrevice, stepOverSteppingStones, climbPastRocks, enterTunnelEntrance,
+		talkToArturHosidius, killSandSnake, searchChest, talkToLordKandurAgain;
+
+	Zone houseWestOfVinery, arceuusLibraryF1, arceuusLibraryF2, arceuusLibraryF3, crabclawCaves, crabclawCavesPart1A,
+		crabclawCavesPart1B, crabclawCavesPart2, crabclawCavesPart3, crabclawCavesDownstairs;
+
+	@Override
+	public Map<Integer, QuestStep> loadSteps()
+	{
+		loadZones();
+		setupRequirements();
+		setupConditions();
+		setupSteps();
+
+		Map<Integer, QuestStep> steps = new HashMap<>();
+
+		ConditionalStep goTalkToLordKandur = new ConditionalStep(this, talkToLordKandur, inVineryHouse);
+		steps.put(0, goTalkToLordKandur);
+
+		ConditionalStep goTalkToChefOlivia = new ConditionalStep(this, talkToChefOlivia, inVineryHouse);
+		steps.put(1, goTalkToChefOlivia);
+
+		ConditionalStep goTalkToGalana = new ConditionalStep(this, talkToGalana, inArceuusLibrary);
+		steps.put(2, goTalkToGalana);
+
+		ConditionalStep findAndReadTheVarlamoreEnvoy = new ConditionalStep(this, findTheVarlamoreEnvoy);
+		findAndReadTheVarlamoreEnvoy.addStep(hasTheVarlamoreEnvoy, readTheVarlamoreEnvoy);
+		findAndReadTheVarlamoreEnvoy.addStep(inArceuusLibrary, talkToGalana);
+		steps.put(3, findAndReadTheVarlamoreEnvoy);
+
+		ConditionalStep stepEnterCrabclawCaves = new ConditionalStep(this, enterCrabclawCaves);
+		steps.put(4, stepEnterCrabclawCaves);
+
+		ConditionalStep stepEnterLowerCaves = new ConditionalStep(this, enterCrabclawCaves);
+		stepEnterLowerCaves.addStep(inFirstAreaOfCaves, goThroughCrevice);
+		stepEnterLowerCaves.addStep(inSecondAreaOfCaves, stepOverSteppingStones);
+		stepEnterLowerCaves.addStep(inThirdAreaOfCaves, climbPastRocks);
+		stepEnterLowerCaves.addStep(inCaves, enterTunnelEntrance);
+		steps.put(6, stepEnterLowerCaves);
+
+		ConditionalStep stepTalkToArturHosidius = new ConditionalStep(this, enterCrabclawCaves);
```

Comment:
Rather than repeating the getting to the lower caves here and for the remaining steps, I'd just re-use the stepEnterLowerCaves ConditionalStep for each of these. For example:

```
ConditionalStep stepTalkToArturHosidius = new ConditionalStep(this, stepEnterLowerCaves);
stepTalkToArturHosidius.addStep(downstairsInCaves, talkToArturHosidius);
```

Same goes for the following steps.

Reply:
I've updated it with what I think you meant. Can you check if that's the correct way? I can't test it anymore, but I can use the same approach in the Grand Tree, I think.